### PR TITLE
[Doc] add overviews for table design

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/table_design/catalog_db_tbl.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## [表](./table_types/table_types.md)
+
+StarRocks 中的表分为两类：内部表和外部表。
+
+**内部表**
+
+内部表归属于 Internal catalog 的数据库，数据保存在 StarRocks 中。内部表由行和列构成，每一行数据是一条记录。
+
+> tip（这个tip可以做一个隐藏展示）
```
Comment (evelynzhaojie) [inline docs/zh/table_design/catalog_db_tbl.md:29]: 这个格式不对，括号的内容删除

File: docs/zh/table_design/catalog_db_tbl.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## [权限系统](../administration/privilege_overview.md)
+
+权限决定了哪些用户可以对哪些特定对象执行哪些特定的操作。StarRocks 采用了两种权限模型：基于用户标识的访问控制和基于角色的访问控制。您可以将权限赋予给角色然后通过角色传递权限给用户，或直接赋予权限给用户标识。
+
+## [存算分离架构下的存储方式](../introduction/Architecture.md#存算分离)
+
+StarRocks 从 3.0 版本开始引入存算分离架构，数据存储功能从原来的 BE 中抽离，数据可持久存储在更为可靠廉价的远端对象存储（如 S3）或 HDFS 上，本地磁盘只用于缓存热数据来加速查询。
+
+云原生表的使用方式兼容存算一体中的内部表。存储卷由远程数据存储系统的属性和凭证信息组成，用于访问存储远端数据。
```
Comment (evelynzhaojie) [inline docs/zh/table_design/catalog_db_tbl.md:57]: 这句话是不是要修改，或者删除。55 行修改后没有再提到云原生表和存储卷了

File: docs/zh/table_design/StarRocks_table_design.md [hunk 3]
```diff
@@ -2,70 +2,136 @@
...
 
-1. 先查找前缀索引表，获得逻辑数据块的起始行号。
-2. 查找维度列的行号索引，定位到维度列的数据块。
-3. 读取数据块。
-4. 解压、解码数据块。
-5. 从数据块中找到维度列前缀对应的数据项。
+## 全面了解表结构
 
-## 加速处理
+本节详细介绍 StarRocks 表结构，帮助您深入了解表结构，根据场景设计高效的数据管理结构。。
```
Comment (evelynzhaojie) [inline docs/zh/table_design/StarRocks_table_design.md:89]: 两个句号

File: docs/zh/table_design/StarRocks_table_design.md [hunk 4]
```diff
@@ -2,70 +2,136 @@
...
+
+除了基本的数据类型，如数值、日期和字符串类型，StarRocks 还支持复杂的半结构化数据类型，包括 ARRAY、JSON、MAP、STRUCT。
+
+### 索引
+
+索引是一种特殊的数据结构，相当于数据的目录。查询条件命中索引列时，StarRocks 能够快速定位到满足条件的数据的位置。
+
+StarRocks 内置一级索引，是根据组成排序键的列的前缀组成的前缀索引。也支持用户手动创建二级索引，以提高查询效率，包括 Bitmap 和 Bloom Filter 索引。
+
+### **约束**
```
Comment (evelynzhaojie) [inline docs/zh/table_design/StarRocks_table_design.md:131]: 这一级不用加粗吧 其他的没加

File: docs/en/table_design/StarRocks_table_design.md [hunk 5]
```diff
@@ -2,74 +2,134 @@
 displayed_sidebar: "English"
 ---
 
-# Understand table design
+# Table overview
 
-## Columnar storage
+Tables are units of data storage. Understanding the table structure in StarRocks and how to design an efficient table structure helps optimize data organization and enhance query efficiency. Also, compared to traditional databases, StarRocks stores complex semi-structured data such as JSON, ARRAY, in a columnar manner to improve query performance.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:7]: StarRocks can store xxx

File: docs/en/table_design/StarRocks_table_design.md [hunk 6]
```diff
@@ -2,74 +2,134 @@
...
+    age int, 
+    phone varchar(16),
+    last_access datetime,
+    credits double
+)
+DUPLICATE KEY(uid, name);
+```
 
-The following figure shows how the StarRocks table design works to accelerate queries on a table in StarRocks.
+The above CREATE TABLE example creates a Duplicate Key table. No constraint is added to columns in this type of table, so duplicate data rows exist in the table. The first two columns of the Duplicate Key table are specified as sort columns to form the sort key. Data is stored after being sorted based on the sort key, which can expedites indexing during queries.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:34]: so duplicate data rows can exist   加个 can

Comment (evelynzhaojie) [reply docs/en/table_design/StarRocks_table_design.md:34]: expedites  -  accelerate

File: docs/en/table_design/StarRocks_table_design.md [hunk 7]
```diff
@@ -2,74 +2,134 @@
...
+"compression" = "LZ4"
+);
+1 row in set (0.00 sec)
+```
 
-2. Search the row number index table of each dimension column to locate the data blocks for the dimension column.
+## Understand comprehensive table structure
 
-3. Read the data blocks.
+A deep dive into StarRocks table structures helps you design efficient data management structure tailored  to your business needs.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:87]: tailored  to --- extra space

File: docs/en/table_design/StarRocks_table_design.md [hunk 8]
```diff
@@ -2,74 +2,134 @@
...
 
-4. Decompress and decode the data blocks.
+### [Table types](./table_types/table_types.md)
 
-5. Search the data blocks to locate the row onto which the dimension column index is mapped.
+StarRocks provides four types of tables which are Duplicate Key tables, Primary Key tables, Aggregate tables, and Unique Key tables, to store data for various business scenarios, such as raw data, frequently updated realtime data, and aggregated data.
 
-## Accelerated processing
+- Duplicate Key tables are simple and easy to use. No constraint is added to columns in this type of table, so duplicate data rows exist in the table. Duplicate Key tables are suitable for storing raw data, such as logs, that does not need any constraints or pre-aggregation.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:93]: can exist

File: docs/en/table_design/catalog_db_tbl.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+A database is a collection of data objects, such as tables, views, and materialized views, to store, manage, and manipulate data.
+
+## [Tables](./table_types/table_types.md)
+
+Tables are categorized into internal tables and external tables.
+
+**Internal tables**
+
+Internal tables are maintained in internal catalogs. The data of internal tables is in StarRocks. Internal tables are maintained in databases under the internal catalog. An internal table consists of rows and columns, where each row represents a record and each column represents an attribute or field of a data row.
```
Comment (evelynzhaojie) [inline docs/en/table_design/catalog_db_tbl.md:27]: is in StarRocks. --- is stored in StarRocks.

File: docs/en/table_design/catalog_db_tbl.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+> Tip (this tip can be hidden/displayed):
```
Comment (evelynzhaojie) [inline docs/en/table_design/catalog_db_tbl.md:29]: same comment as in Chinese

File: docs/en/table_design/catalog_db_tbl.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+## [Materialized views](../using_starrocks/Materialized_view.md)
+
+Materialized views are special physical tables that hold precomputed query results from one or more base tables. Materialized views are categorized as synchronous and asynchronous materialized views. Asynchronous materialized views are more powerful because they can store precomputed results from multiple base tables (internal and external tables) and support various aggregation operators. When complex queries are executed on base tables, StarRocks can automatically reuse precomputed results stored in materialized views, to achieve transparent query acceleration, accelerate queries in data lakes and establish data modeling.
+
+## [Views](../sql-reference/sql-statements/data-definition/CREATE_VIEW.md)
+
+Views, or logical views, are virtual tables that do not hold any data. Views only display data that is the results of queries against other existing physical tables. The query that defines a view is run every time the view is referenced in a query.
+
+## [Permission system](../administration/privilege_overview.md)
```
Comment (evelynzhaojie) [inline docs/en/table_design/catalog_db_tbl.md:49]: Privilege

File: docs/en/table_design/catalog_db_tbl.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+Privileges determine which users can perform which operations on which objects. StarRocks adopts two types of permission models: identity-based access control and role-based access control. You can first assign privileges to roles, and then assign roles to users. In this case, privileges are passed to users through roles. Or, you can directly assign privileges to user identities.
+
+## [Data storage in storage-compute separation architecture](../introduction/Architecture.md#storage-compute-separation)
+
+Since v3.0, StarRocks introduces the new storage-compute separation (shared-data) architecture. Data storage is separated from BEs. Data is persistently stored in remote object storage or HDFS, while local disks are used for caching hot data to accelerate queries. 
+
+The usage of cloud-native tables is compatible with that of internal tables in the storage-compute coupled architecture. A storage volume consists of the properties and credential information of the remote data storage, enabling access to data stored remotely.
```
Comment (evelynzhaojie) [inline docs/en/table_design/catalog_db_tbl.md:57]: update with chinese

File: docs/en/table_design/StarRocks_table_design.md [hunk 13]
```diff
@@ -2,74 +2,134 @@
...
+
+StarRocks provides a built-in primary index, which is a prefix index composed of the prefixes of the table's sort key columns. StarRocks also allows you to create secondary indexes, including Bitmap and Bloom Filter indexes, by hand to further enhance query efficiency, .
+
+### Constraints
+
+Constraints help ensure data integrity, consistency, and accuracy. The primary key columns in Primary Key tables are constrained to have unique and NOT NULL values. The aggregate key columns in Aggregate tables and the unique key columns in Unique Key tables are constrained to have unique values.
+
+### More features
+
+Apart from the above features, you can adopt more features based on your business requirements to design a more robust table structure. For example, using Bitmap and HLL columns to accelerate deduplication counting, specifying generated columns or auto-increment columns to speed up some queries, configuring flexible and automatic mechanism for storage cooldown to reduce maintainance costs, and configuring Colocate Join to speed up multi-table JOIN queries. For more details, see [CREATE TABLE](../sql-reference/sql-statements/data-definition/CREATE_TABLE.md).
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:135]: 1. deduplication counting --- distinct counting    2. configuring flexible and automatic storage cooldown methods to xxx

File: docs/en/table_design/StarRocks_table_design.md [hunk 14]
```diff
@@ -2,74 +2,134 @@
...
+
+### Index
+
+An index is a special data structure and is used as a pointer to data in a table. When the conditional columns in queries are indexed columns, StarRocks can swiftly locate the data that meets the conditions.
+
+StarRocks provides a built-in primary index, which is a prefix index composed of the prefixes of the table's sort key columns. StarRocks also allows you to create secondary indexes, including Bitmap and Bloom Filter indexes, by hand to further enhance query efficiency, .
+
+### Constraints
+
+Constraints help ensure data integrity, consistency, and accuracy. The primary key columns in Primary Key tables are constrained to have unique and NOT NULL values. The aggregate key columns in Aggregate tables and the unique key columns in Unique Key tables are constrained to have unique values.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:131]: The primary key columns in Primary Key tables must have unique and NOT NULL values. The aggregate key columns in Aggregate tables and the unique key columns in Unique Key tables must have unique values.

File: docs/en/table_design/StarRocks_table_design.md [hunk 15]
```diff
@@ -2,74 +2,134 @@
...
+### [Data types](../sql-reference/sql-statements/data-types/data-type-list.md)
 
-- A bitmap index is used to locate the rows that meet specified query conditions in a column of the ENUM data type.
+In addition to basic data types such as NUMERIC, DATE, and STRING, StarRocks supports complex semi-structured data types, including ARRAY, JSON, MAP, and STRUCT.
+
+### Index
+
+An index is a special data structure and is used as a pointer to data in a table. When the conditional columns in queries are indexed columns, StarRocks can swiftly locate the data that meets the conditions.
+
+StarRocks provides a built-in primary index, which is a prefix index composed of the prefixes of the table's sort key columns. StarRocks also allows you to create secondary indexes, including Bitmap and Bloom Filter indexes, by hand to further enhance query efficiency, .
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:127]: delete "by hand"

File: docs/en/table_design/StarRocks_table_design.md [hunk 16]
```diff
@@ -2,74 +2,134 @@
...
+#### Bucketing
 
-### Per-column index
+The second level is bucketing: Data within a partition is further divided into smaller data management units through bucketing. Replicas of each bucket are evenly distributed across BEs to ensure high data availability.
 
-StarRocks supports per-column indexes such as Bloom filters, zone maps, and bitmap indexes:
+StarRocks provides two bucketing methods:
 
-- A Bloom filter is used to determine whether data blocks contain the values you want to query.
+- Hash bucketing: Data is distributed into buckets based on the hash values of the bucketing key. When the bucketing key consists of columns that are frequently used as the conditional columns in queries, query efficiency can be greatly improved.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:116]: 第二句意思不太对，You can select columns that are frequently used as condition columns in queries as bucketing columns, which helps improve query efficiency.

File: docs/en/table_design/StarRocks_table_design.md [hunk 17]
```diff
@@ -2,74 +2,134 @@
...
+StarRocks provides four types of tables which are Duplicate Key tables, Primary Key tables, Aggregate tables, and Unique Key tables, to store data for various business scenarios, such as raw data, frequently updated realtime data, and aggregated data.
 
-## Accelerated processing
+- Duplicate Key tables are simple and easy to use. No constraint is added to columns in this type of table, so duplicate data rows exist in the table. Duplicate Key tables are suitable for storing raw data, such as logs, that does not need any constraints or pre-aggregation.
+- Primary Key tables are powerful. Both unique and non-null constraints are added to primary key columns. Primary Key tables support real-time frequent updates and partial column updates, while ensuring high query performance, and therefore are suitable for real-time queries scenarios.
+- Aggregate tables are suitable to store pre-aggregated data, helping reduce the amount of data scanned and calculated and improve efficiency for aggregation queries.
+- Unique tables are also suitable to store frequently updated realtime data. However this type of tables is being replaced by Primary Key tables, which are more powerful.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:96]: suitable for storing

File: docs/en/table_design/StarRocks_table_design.md [hunk 18]
```diff
@@ -2,74 +2,134 @@
...
 
-4. Decompress and decode the data blocks.
+### [Table types](./table_types/table_types.md)
 
-5. Search the data blocks to locate the row onto which the dimension column index is mapped.
+StarRocks provides four types of tables which are Duplicate Key tables, Primary Key tables, Aggregate tables, and Unique Key tables, to store data for various business scenarios, such as raw data, frequently updated realtime data, and aggregated data.
 
-## Accelerated processing
+- Duplicate Key tables are simple and easy to use. No constraint is added to columns in this type of table, so duplicate data rows exist in the table. Duplicate Key tables are suitable for storing raw data, such as logs, that does not need any constraints or pre-aggregation.
+- Primary Key tables are powerful. Both unique and non-null constraints are added to primary key columns. Primary Key tables support real-time frequent updates and partial column updates, while ensuring high query performance, and therefore are suitable for real-time queries scenarios.
```
Comment (evelynzhaojie) [inline docs/en/table_design/StarRocks_table_design.md:94]: Both unique and non-null constraints are added to primary key columns. --- not easy to understand. The primary key columns must have unique and NOT NULL values.

**Review by evelynzhaojie — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Pull request merged**